tests/virt_autotest/login_console_as_kvm_or_xen_role.pm

```diff
@@ -0,0 +1,66 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: For openSUSE virtualization test only. login in console, install kvm/xen patterns if needed.
+# Maintainer: Julie CAO <jcao@suse.com>
+
+use strict;
+use warnings;
+use base 'virt_autotest_base';
+use testapi;
+use ipmi_backend_utils;
+use Utils::Backends qw(use_ssh_serial_console is_remote_backend);
+use utils qw(zypper_call systemctl);
+use virt_autotest::utils qw(is_kvm_host is_xen_host);
+
+sub run {
+
+    #local openQA can skip TW host install
+    if (get_var('IPMI_DO_NOT_RESTART_HOST')) {
+        select_console 'sol', await_console => 0;
+        send_key 'ret';
+    }
+
+    use_ssh_serial_console !check_screen('text-logged-in-root');
+    return if get_var('IPMI_DO_NOT_RESTART_HOST');
+
+    zypper_call("--gpg-auto-import-keys ref",         180);
```

Comment:
I suggest to just keep the default timeout unless you want to enlarge it. Just use the default which is 700s. Also I recommend to use single-ticks for quoting unless you want to expand variables in strings

Reply:
No problem. I just copied this line elsewhere and did not think much.